File: server/models/mongo/supervisor.model.js
import mongoose from "mongoose";
const Schema = mongoose.Schema;

const supervisorSchema = new Schema({
    supervisor_id: {
        type: Map,
        of: String,
        required: true
    },
    name: {
        type: Map,
        of: String,
        required: true
    },
    password: {
        type: Map,
        of: String,
        required: true
    },
    phone_no: {
        type: Map,
        of: String,
        required: true,
    },
    email: {
        type: Map,
        of: String,
        required: true
    },
    date_of_birth: {
        type: Map,
        of: String,
        required: true
    },
    age: {
        type: Map,
        of: String,
        required: true
    },
    sex: {
        type: Map,
        of: String,
        required: true,
        enum: ['M', 'F', 'O']
    },
    department: {
        type: Map,
        of: String,
        required: true
    },
    preferred_language1: {
        type: Map,
        of: String,
        required: true
    },
    preferred_language2: {
        type: Map,
        of: String,
        required: true
    },
    preferred_language3: {
        type: Map,
        of: String,
        required: true
    },
    user_image: {
        type: Map,
        of: String,
    },
    supervisor_id_hash: {
        type: String,
        required: true
    },
    email_hash: {
        type: String,
        required: true
    },
    phone_hash: {
        type: String,
        required: true
    },
    authenticated: {
        type: String,
        required: true,
        default: false
    },
    allocated_therapists: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: "student_therapist",
            default: null,
        }
    ],
    allocated_patients_active: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: "patient",
            default: null,
        }
    ],
    allocated_patients_inactive: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: "patient",
            default: null,
        }
    ],
    reports_assessed: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: "report",
            default: null,
        }
    ]
});

export const Supervisor = mongoose.model("supervisor", supervisorSchema);

File: frontend/flutter/lib/Screens/Supervisor/Screens/model/supervisor_model.dart
class Supervisor {
  final String id;
  final String supervisorId;
  final String name;
  final String email;
  final String phoneNo;
  final String department;
  final String preferredLanguage1;
  final String preferredLanguage2;
  final String preferredLanguage3;

  Supervisor({
    required this.id,
    required this.supervisorId,
    required this.name,
    required this.email,
    required this.phoneNo,
    required this.department,
    required this.preferredLanguage1,
    required this.preferredLanguage2,
    required this.preferredLanguage3,
  });

  factory Supervisor.fromJson(Map<String, dynamic> json) {
    return Supervisor(
      id: json['_id'] ?? "",
      supervisorId: json['supervisor_id'] ?? "",
      name: json['name'] ?? "",
      email: json['email'] ?? "",
      phoneNo: json['phone_no'] ?? "",
      department: json['department'] ?? "",
      preferredLanguage1: json['preferred_language1'] ?? "",
      preferredLanguage2: json['preferred_language2'] ?? "",
      preferredLanguage3: json['preferred_language3'] ?? "",
    );
  }

   Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'phone_no': phoneNo,
      'department': department,
      'preferred_language1': preferredLanguage1,
      'preferred_language2': preferredLanguage2,
      'preferred_language3': preferredLanguage3,
    };
   }
}
